### ISSUE TITLE
dgc ta sign Empty input

### ISSUE THREAD

**Issue by IvarsAkmentins** [eu-digital-green-certificates/dgc-cli#4]
Hi, 
I created self signed certificate (`openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365`)
and trust anchor (`keytool -importcert -keystore trustanchor.jks -storepass password -file your-created-trustanchor-cert.pem -alias trustanchor`).

I try to sign generated certificate with trust anchor (`dgc ta sign -c trustanchor.jks -k trust-anchor-key.pem -i your-cert.crt`) and get error message:

> java.security.cert.CertificateException: Could not parse certificate: java.io.IOException: Empty input
        at java.base/sun.security.provider.X509Factory.engineGenerateCertificate(X509Factory.java:115)
        at java.base/java.security.cert.CertificateFactory.generateCertificate(CertificateFactory.java:355)
        at eu.europa.ec.dgc.cli.utils.CliUtils.readCertFromFile(CliUtils.java:67)
        at eu.europa.ec.dgc.cli.trustanchor.Sign.call(Sign.java:82)
        at eu.europa.ec.dgc.cli.trustanchor.Sign.call(Sign.java:47)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
        at picocli.CommandLine.execute(CommandLine.java:2078)
        at eu.europa.ec.dgc.cli.Main.main(Main.java:28)
Caused by: java.io.IOException: Empty input
        at java.base/sun.security.provider.X509Factory.engineGenerateCertificate(X509Factory.java:111)
        ... 12 more

Do I have to change the certificate in different format?

Thank you.

**Comment by f11h** [comment]
Hi @IvarsAkmentins ,

Yes you have to provide the Trustanchor Public Key in PEM Format. I will keep in mind to make it possible to use JKS (or at least Print Out a more helpfull Error message)

**Comment by IvarsAkmentins** [comment]
Hi,

thank you for response, but I still have errors. Just to clarify for `dgc ta sign` options i have:

`--trustAnchorCert` = here i use trust anchor certificate file in pem format
`-- inputCert` = certificate pem file which i want to sign
`-- trustAnchorKey` = trust anchore public key in pem format

Is this correct?

**Comment by f11h** [comment]
yes thats correct. At the moment dgc-cli only supports unencrypted private keys. Maybe you have set a password when creating it

**Comment by IvarsAkmentins** [comment]
It did not work for me, I unencrypted the trust anchor private key with
`openssl rsa -in <Encrypted key filename>  -out < desired output file name>`

`called dgc` ta sign and got response `java.lang.IllegalArgumentException: unknown object in getInstance: org.bouncycastle.openssl.PEMKeyPair`

---
Steps I just reproduced
1. create TA self signed certificate `openssl req -x509 -newkey rsa:4096 -keyout ta-key.pem -out ta-cert.pem -days 365`
2. create self signed certificate I want to sign later with ta `openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365`
3. unencrypted ta-key.pem with `openssl rsa -in ta-key.pem  -out ta-unenc.pem`
4. called `dgc ta sign -c ./ta-cert.pem -k ta-unenc.pem -i ./cert.pem`

**Comment by f11h** [comment]
Hi @IvarsAkmentins ,

I found the reason for this problem. It is because unencrypting the private key with ```openssl rsa``` will output an PKCS#1 format key. DGC-CLI does not support this at the moment.

I filed a bug for this --> #9 

At the moment I have two workarounds for this problem:
1. Create the PrivateKey unencrypted:
```
openssl req -x509 -newkey rsa:4096 -keyout ta-key-unenc.pem -out ta-cert.pem -days 365 -nodes
```
2. Unencrypt the key with the following command:
```
openssl pkcs8 -topk8 -inform PEM -outform PEM -nocrypt -in ta-key.pem -out ta-key-unenc.pem
```
